test/specs/core.plugin.tests.js

```diff
@@ -71,38 +8,42 @@ describe('Chart.plugins', function() {
 
 			spyOn(plugin, 'hook');
 
-			Chart.plugins.notify(chart, 'hook', 42);
+			chart._plugins.notify(chart, 'hook', 42);
 			expect(plugin.hook.calls.count()).toBe(1);
 			expect(plugin.hook.calls.first().args[0]).toBe(chart);
 			expect(plugin.hook.calls.first().args[1]).toBe(42);
 			expect(plugin.hook.calls.first().args[2]).toEqual({});
 		});
 
 		it('should call global plugins with arguments', function() {
-			var plugin = {hook: function() {}};
+			var plugin = {id: 'a', hook: function() {}};
 			var chart = window.acquireChart({});
 
 			spyOn(plugin, 'hook');
 
-			Chart.plugins.register(plugin);
-			Chart.plugins.notify(chart, 'hook', 42);
+			Chart.register(plugin);
+			chart.update();
```

Comment:
why do we need to update the chart now?

Reply:
Registering a plugin does not invalidate the descriptors, and so the new plugin is not noticed before update.
Maybe its better to have the registering invalidate the descriptors like before. Though changing anything else needs an `update`, so it would be more consistent this way.
Opinions?

Reply:
Depends what you mean by "changing anything else". If you just mean changing the chart options, then I think I'd prefer to have registering invalidate the descriptors to keep things like before. But let me know if there's something else you had in mind that I'm not considering

Reply:
Yeah, I was thinking of changing chart options or defaults.
Changed it to invalidate when registering / unregistering something (not only plugins).